src/deepsparse/loggers/helpers.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Optional, Tuple
+
+from pydantic import BaseModel
+
+
+"""
+Helpers functions for logging
+"""
+
+__all__ = ["possibly_extract_value", "match"]
+
+
+def possibly_extract_value(
+    value: Any, identifier_reminder: Optional[str] = None
+) -> Any:
+    """
+    Given a string of ("."-separated) sub_identifiers, access the items
+    inside `value`.
+
+    :param value: A data structure that may potentially hold "nested" values
+        of interest.
+    :param sub_identifiers: A string of "."-separated keys that are used to
+        access "nested" value of interest inside`value`.
+    :return: Value of interest
+    """
+    if not identifier_reminder:
+        return value
+
+    value = dict(value) if isinstance(value, BaseModel) else value
+
+    for reminder in identifier_reminder.split("."):
+        # TODO: Add the support for slicing
+        # e.g. pipeline_inputs.images[0,0,0,0]
+        value = value[reminder]
+    return value
+
+
+def match(template: str, identifier: str) -> Tuple[bool, Optional[str]]:
+    """
+    Match the template against the identifier
+
+    :param template: A string in format:
+        <pipeline_name>.<target_name>.<reminder>
+        or
+        <target_name>.<reminder>
```

Comment:
we should support more than this as this constrains what the pipelines can log (ie nested pipelines may be logging multiple levels of pipeline or target names). let's instead support `<anything_1>...<anything_n><remainder>`

we'd have 3 cases

```
template == identifier -> return True, value
template.startswith(identifier) -> split template into the identifier + remainder, then extract remainder
template starts with our regex key -> return True if it's a regex match
return False otherwise 
```

Reply:
^ one thing here is we should have a prefix the user uses to denote they are using regex, otherwise regex will always match to identifiers that contain entire targets

Reply:
There is one edge case that is not being taken into account here.
Imagine we are in the non-server context.

The user creates an image classification pipeline and specifies the loggers as follows:

```python
logger = FunctionLogger(logger = PythonLogger(), identifier = "pipeline_inputs.images", ...)
pipeline = Pipeline.create(..., logger)
```
Their goal is to apply some function to the `images` in `pipeline_inputs`. Thus:
`self.template_identifier = "pipeline_inputs.images`

However, what is being sent to the function logger from the pipeline at every forward pass is:
`identifier = "image_classification.pipeline_inputs.images`

Okay, we see that the matching should output `match = True, remainder = "images"`.
But for every one of the cases you mentioned this will not happen:
1. will fail
2. will fail
3. will probably also fail

We need to be able to assume, that in the non-server context, the FunctionLogger will match templates that do not know about the pipeline name (it is not included in the template), with the identifiers that do contain the pipeline name.

Reply:
On second thought, I am okay with neglecting the non-server implementation and cutting corners here. Let's not worry about the non-server scenario for now and focus on the iteration speed.